pkg/kubeadm/master/manifests.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubemaster
+
+import (
+	"bytes"
+	"encoding/json"
+	_ "fmt"
+	"os"
+	"path"
+
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	api "k8s.io/kubernetes/pkg/api/v1"
+	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/util/intstr"
+)
+
+// Static pod definitions in golang form are included below so that `kubeadm
+// init master` and `kubeadm manual bootstrap master` can get going.
+
+const (
+	COMPONENT_LOGLEVEL       = "--v=4"
+	SERVICE_CLUSTER_IP_RANGE = "--service-cluster-ip-range=10.16.0.0/12"
+	CLUSTER_NAME             = "--cluster-name=kubernetes"
+	MASTER                   = "--master=127.0.0.1:8080"
+	HYPERKUBE_IMAGE          = "errordeveloper/hyperquick:master"
+)
+
+// TODO look into what this really means, scheduler prints it for some reason
+//
+//E0817 17:53:22.242658       1 event.go:258] Could not construct reference to: '&api.Endpoints{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"kube-scheduler", GenerateName:"", Namespace:"kube-system", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]api.OwnerReference(nil), Finalizers:[]string(nil)}, Subsets:[]api.EndpointSubset(nil)}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'Normal' '%v became leader' 'moby'
+
+func WriteStaticPodManifests(params *kubeadmapi.BootstrapParams) error {
+	staticPodSpecs := map[string]api.Pod{
+		// TODO this needs a volume
+		"etcd": componentPod(api.Container{
+			Command: []string{
+				"/usr/local/bin/etcd",
+				"--listen-peer-urls=http://127.0.0.1:2380",
+				"--addr=127.0.0.1:2379",
+				"--bind-addr=127.0.0.1:2379",
+				"--data-dir=/var/etcd/data",
+			},
+			Image:         "gcr.io/google_containers/etcd:2.2.1", // TODO parametrise
```

Comment:
Let's go with 2.2.5 for now; but please make that configurable too.
Also change to `binary-arch:version` naming and default arch to `GOARCH`